server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -544,6 +547,16 @@ private IndexMetadata buildAndValidateTemporaryIndexMetadata(
         tmpImdBuilder.settings(indexSettings);
         tmpImdBuilder.system(isSystem);
 
+        if (isRemoteStoreAttributePresent(settings)) {
+            String pathType;
+            if (clusterService.getClusterSettings().get(IndicesService.CLUSTER_REMOTE_STORE_PATH_PREFIX_OPTIMISED_SETTING)) {
+                pathType = RemoteStorePathType.HASHED_PREFIX.toString();
+            } else {
+                pathType = RemoteStorePathType.FIXED.toString();
+            }
+            tmpImdBuilder.putCustom(IndexMetadata.REMOTE_STORE_CUSTOM_KEY, Map.of(RemoteStorePathType.NAME, pathType));
+        }
+
```

Comment:
Can we encapsulate this logic in `RemoteStoreBlobPathFactory`

Reply:
I have tried to address comments. I will add a factory in following PR that will be used in places wherever we want to derive the path which we depend on the `RemoteStoreBlobPathType`.